app/concerns/liquid_droppable.rb

```diff
@@ -82,9 +84,49 @@ def to_liquid
     end
   end
 
+  class ActiveRecordCollectionDrop < Drop
+    include Enumerable
+
+    def each(&block)
+      @object.each(&block)
+    end
+
+    # required for variable indexing as array
+    def [](i)
+      case i
+      when Integer
+        @object[i]
+      when 'size', 'first', 'last'
+        __send__(i)
+      end
+    end
+
+    # required for variable indexing as array
+    def fetch(i, &block)
+      @object.fetch(i, &block)
+    end
+
+    # compatibility with array; also required by the `size` filter
+    def size
+      @object.count
+    end
+
+    # compatibility with array
+    def first
+      @object.first
+    end
+
+    # compatibility with array
+    def last
+      @object.last
+    end
+
+    # This drop currently does not support the `slice` filter.
```

Comment:
This worries me a bit because it's a breaking change, even though I am not sure how useful `silce` is on `sources`, `receivers` etc.

Reply:
The [`slice`](https://github.com/Shopify/liquid/blob/27c91203ab4adfde2cc5e1142bce2a267634ef8e/lib/liquid/standardfilters.rb#L52-L61) filter currently does not accept general Enumerable objects and I wasn't sure either if it's worth patching the `slice` filter, so I left it out. 🤔

Reply:
I guess that is because `slice` is only implemented for `Array` and `String`. Let's see if anyone complains about it, since the associations are not ordered only mysql user could have used `slice`.